core/trino-main/src/main/java/io/trino/operator/OperatorStats.java

```diff
@@ -498,6 +509,7 @@ public OperatorStats add(Iterable<OperatorStats> operators)
             outputPositions += operator.getOutputPositions();
 
             dynamicFilterSplitsProcessed += operator.getDynamicFilterSplitsProcessed();
+            metrics = Metrics.merge(metrics, operator.getMetrics());
```

Comment:
I think one disadvantage of working with Metrics wrapper is that it would result in the underlying map getting copied many times here.
It might be more efficient to work directly with the underlying Map for merging here
E.g.
```
Map<> metricsAccumulator = this.getMetrics().getMetrics()
...
for (OperatorStats operator : operators) {
....
    operator.getMetrics().forEach((key, value) -> metricsAccumulator.merge(key, value, (m1, m2) -> m1.mergeWith(m2))));
....
}
Metrics merged = new Metrics(metricsAccumulator);

```

Reply:
Sounds good - created an helper class for accumulating metrics (similar to the "builder" pattern).